### ISSUE TITLE
[Bug]: cmp + show_settings error

### ISSUE THREAD

**Issue by sQVe** [olimorris/codecompanion.nvim#224]
### Your `minimal.lua` config

```lua
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- set config path to enable the copilot adapter to work
vim.env["CODECOMPANION_TOKEN_PATH"] = vim.fn.expand("~/.config")

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({
    "git",
    "clone",
    "--filter=blob:none",
    "--single-branch",
    "https://github.com/folke/lazy.nvim.git",
    lazypath,
  })
end
vim.opt.runtimepath:prepend(lazypath)

-- Your CodeCompanion setup
local plugins = {
  {
    "olimorris/codecompanion.nvim",
    dependencies = {
      { "nvim-treesitter/nvim-treesitter", build = ":TSUpdate" },
      "nvim-lua/plenary.nvim",
      "hrsh7th/nvim-cmp",
      {
        "stevearc/dressing.nvim",
        opts = {},
      },
      "nvim-telescope/telescope.nvim",
    },
    opts = {
      -- Refer to: https://github.com/olimorris/codecompanion.nvim/blob/main/lua/codecompanion/config.lua
      strategies = { -- Change the adapters as required
        chat = { adapter = "openai" },
        inline = { adapter = "openai" },
        agent = { adapter = "openai" },
      },
      opts = {
        log_level = "DEBUG",
      },

      display = {
        chat = { show_settings = true } }
    },
  }
}

require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

-- Setup Tree-sitter
local ok, treesitter = pcall(require, "nvim-treesitter.configs")
if ok then
  treesitter.setup({
    ensure_installed = { "lua", "markdown", "markdown_inline", "yaml" },
    highlight = { enable = true },
  })
end

-- Setup completion
local ok, cmp = pcall(require, "cmp")
if ok then
  cmp.setup({
    mapping = cmp.mapping.preset.insert({
      ["<C-b>"] = cmp.mapping.scroll_docs(-4),
      ["<C-f>"] = cmp.mapping.scroll_docs(4),
      ["<C-Space>"] = cmp.mapping.complete(),
      ["<C-e>"] = cmp.mapping.abort(),
      ["<CR>"] = cmp.mapping.confirm({ select = true }), -- Accept currently selected item. Set `select` to `false` to only confirm explicitly selected items.
    }),
  })
end
```

### Error messages

```text
Error detected while processing TextChangedI Autocommands for "*":                                                                                                         
Error executing lua callback: ...gins/codecompanion.nvim/lua/cmp_codecompanion/models.lua:23: attempt to index field 'chat' (a nil value)                                  
stack traceback:                                                                                                                                                           
        ...gins/codecompanion.nvim/lua/cmp_codecompanion/models.lua:23: in function 'complete'                                                                             
        /home/sqve/.repro/plugins/nvim-cmp/lua/cmp/source.lua:326: in function 'complete'                                                                                  
        /home/sqve/.repro/plugins/nvim-cmp/lua/cmp/core.lua:299: in function 'complete'                                                                                    
        /home/sqve/.repro/plugins/nvim-cmp/lua/cmp/core.lua:169: in function 'callback'                                                                                    
        /home/sqve/.repro/plugins/nvim-cmp/lua/cmp/core.lua:229: in function 'autoindent'                                                                                  
        /home/sqve/.repro/plugins/nvim-cmp/lua/cmp/core.lua:161: in function 'on_change'                                                                                   
        /home/sqve/.repro/plugins/nvim-cmp/lua/cmp/init.lua:346: in function 'callback'                                                                                    
        ...e/sqve/.repro/plugins/nvim-cmp/lua/cmp/utils/autocmd.lua:49: in function 'emit'                                                                                 
        ...e/sqve/.repro/plugins/nvim-cmp/lua/cmp/utils/autocmd.lua:23: in function <...e/sqve/.repro/plugins/nvim-cmp/lua/cmp/utils/autocmd.lua:22> 
```

### Log output

Not relevant

### Health check output

 ```text
codecompanion: require("codecompanion.health").check()                                                           
                                                                                                                 
- Neovim version: 0.10.1                                                                                         
- Log file: /home/sqve/.repro/state/nvim/codecompanion.log                                                       
                                                                                                                 
Plugins:                                                                                                         
- OK plenary.nvim installed                                                                                      
- OK nvim-treesitter installed                                                                                   
- OK nvim-cmp installed                                                                                          
- OK telescope.nvim installed                                                                                    
- OK dressing.nvim installed                                                                                     
                                                                                                                 
Tree-sitter parsers:                                                                                             
- OK markdown parser installed                                                                                   
- OK yaml parser installed                                                                                       
                                                                                                                 
Libraries:                                                                                                       
- OK curl installed
```

### Describe the bug

As soon as I enable `show_settings` the chat buffer errors, which I assume is related to cmp. It works fine when the settings are turned off.

### Reproduce the bug

1. Use `cmp`
2. Enable `show_settings`
3. Start chat.
4. Type and get error.

### Final checks

- [X] I have made sure this issue exists in the latest version of the plugin
- [X] I have tested with the `minimal.lua` file from above and have shared this
- [X] I have shared the contents of the log file

**Comment by S1M0N38** [comment]
I can confirm. I can reproduce this bug with the following config
```lua
---@diagnostic disable: missing-fields

vim.env.LAZY_STDPATH = ".repro"
load(vim.fn.system("curl -s https://raw.githubusercontent.com/folke/lazy.nvim/main/bootstrap.lua"))()


-- Setup codecompanion
local adapter = {
  adapter = "copilot",
}

local plugins = {
  {
    "olimorris/codecompanion.nvim",
    dependencies = {
      { "nvim-treesitter/nvim-treesitter", build = ":TSUpdate" },
      { "nvim-lua/plenary.nvim" },
      { "hrsh7th/nvim-cmp" },
      { "stevearc/dressing.nvim",          opts = {} },
      { "nvim-telescope/telescope.nvim" },
    },
    opts = {
      strategies = {
        chat = adapter,
        inline = adapter,
        agent = adapter,
      },
      display = {
        chat = {
          show_settings = true,
        },
      },
    },
  },
}

require("lazy.minit").repro({ spec = plugins })

-- Setup Tree-sitter
local ts_status, treesitter = pcall(require, "nvim-treesitter.configs")
if ts_status then
  treesitter.setup({
    ensure_installed = { "lua", "markdown", "markdown_inline", "yaml" },
    highlight = { enable = true },
  })
end

-- Setup completion
local cmp_status, cmp = pcall(require, "cmp")
if cmp_status then
  cmp.setup({
    mapping = cmp.mapping.preset.insert({
      ["<C-b>"] = cmp.mapping.scroll_docs(-4),
      ["<C-f>"] = cmp.mapping.scroll_docs(4),
      ["<C-Space>"] = cmp.mapping.complete(),
      ["<C-e>"] = cmp.mapping.abort(),
      ["<CR>"] = cmp.mapping.confirm({ select = true }),
    }),
  })
end

--Open new CodeCompanion chat
vim.cmd("CodeCompanionChat")
```

**Comment by olimorris** [comment]
Thanks for raising this. I'll send the fix over with the v3.0.0 release tonight.
